src/icp_canister_backend/src/lib.rs

```diff
@@ -156,6 +156,7 @@ fn add_deposit(deposit_id: u64, deposit: types::Deposit, user: Principal, assets
         let mut episode = episodes_ref.get(&deposit.episode).unwrap_or(Episode {
             episode_shares: Nat::from(0u64),
             assets_staked: Nat::from(0u64),
+            is_processed: false, 
```

Comment:
This field is not needed. Instead, make a new stable variable.
LAST_TIME_UPDATED, it should store the last time process_episodes function was called, thus it is possible to calculate what was the last processed episode.